### ISSUE TITLE
assets finally solved

### ISSUE THREAD

**Issue by jimgwhit** [nova-framework/framework#698]
@daveismyname  and @LuckyCyborg  After seeing issue #697, I checked my fileinfo in my php.ini, it was commented, all this time I did not know that. I un-commented it so now all is working out of the box. You see @LuckyCyborg that other stuff you said like a vh or htaccess had nothing to do with it.  
However some users may be on a shared host where they cannot un-comment fileinfo on the host.  
The only change that is needed (and it doesn't break nothing) is changing one line in the framework:
This
```
public static function templatePath($custom = TEMPLATE, $folder = '/assets/')
    {
        $template = Inflector::tableize($custom);

        return SITEURL .'templates/' .$template .$folder;
    }
``` 
TO
```
public static function templatePath($custom = TEMPLATE, $folder = '/assets/')
    {
        $template = Inflector::tableize($custom);

        return SITEURL .'app/templates/' .$template .$folder;  // THIS LINE
    }
```
This is only in the case where you cannot use fileinfo on a shared host.  I suspect this would be rare.  
This is true weather virtual host, above public, An install exactly as @LuckyCyborg would install, doesn't matter, without fileinfo the above works.  I just spent 5 hours testing it with and without fileinfo.  So perhaps a one line blurp about this in the docs wouldn't hurt, it might help someone get setup and going.  
But again most host should have fileinfo.

**Comment by LuckyCyborg** [comment]
@jimgwhit I wonder if you read the framework's **README.md** where is specified very clear into **Requirements** :

> fileinfo enabled (edit php.ini and uncomment php_fileinfo.dll or use php selector within cpanel if available.

All my statements supposed that you respect the framework requirements...

**Comment by jimgwhit** [comment]
@LuckyCyborg what in the world are you even talking about?

**Comment by LuckyCyborg** [comment]
@jimgwhit Your workaround definitively does NOT work in a (Linux based) server using the recommended way of installation (framework above webroot).

> Frankly I think a real instead of a virtual path would have been better anyway

Because you are stuck in **SMVC v2.2** Era. The virtual paths are choose from flexibility and multi-tenant ability reasons. For example, the Routing File Serving works just fine even you rename your "app" as "happydogs" and is located somewhere out of framework tree.

In other hand, please understand that the single major change on Nova (v3), comparative with SMVC  v2.2 is the Routing and structure design to work above the webroot.

Then, I strongly recommend you to stuck into SMVC 2.2 if you really need to use an "all in web-root" way, which is the default way of use for previous version.

Also, note that using the Nova (v3) exposing the framework into webroot, can introduce sever security issues, the framework not being designed for that, then its structure not being shielded at all. 

So, if you insist for using a web exposed way, I strongly recommend you, also from a Security POV, the previous version.

**Comment by jimgwhit** [comment]
@LuckyCyborg  putting it above the Webroot thing I don't know who came up with that junk. The reality with proper security you are fine putting everything under HTDOCS. I have one laravel installation on GoDaddy set up that way but I don't use the env file I hard code the configs and everything is working fine. I honestly believe someone probably some young new college graduate  said it all would it be more secure to do it this way it caught on that everybody and their brother all of a sudden wants to do it that way that doesn't mean it's better or correct. But yes I have another installation where I'm above htdocs.  
The thing is if someone truly hacks into a server well they are pretty well much anywhere they want to be on that server so does it really really really make a difference what folder you have something in if it's hacked it's hacked.  
How come you believe there is a difference from one folder to another folder on a server?

**Comment by LuckyCyborg** [comment]
@jimgwhit Come on! Then those who develop Laravel, CakePHP, CodeIgniter & Co. are "some young new college graduate" and you are a smart-ass, even you still don't known what are the HTTP response codes?

To quit that funny debate, let's say that Dave chosed this way; the version three (Nova) to live above webroot. If you don't like this design, you are free to stuck on version 2.2, which is almost identical, after all, thanks in part also to your own lamentations...

**Comment by Kingorgg** [comment]
@LuckyCyborg It is @jimgwhit's choice what he does with the framework, just leave him be. **_If_** something goes wrong then he has to fix it...

**Comment by jimgwhit** [comment]
@Kingorgg Nothing will go wrong, because @daveismyname does have the framework designed to install either way, that's even in the video.  Laravel can also be installed either way.  I just think @LuckyCyborg is still upset with @daveismyname for not using the super big giant monstor he came up with.  Don't get me wrong I appreciate all the work he put into it, and even the dbal implementation.  But if you go back to the issues right after the now new nova came out, His tone was one of being upset.  And @LuckyCyborg I wish you could clearly understand that there is nothing wrong with my workaround for someone that can't access fileinfo.  Why can't you admit that. 
Both you and I once agreed on dbal, Dave decided not to.  But it can still be used.  And again I use the framework for databasing, but try to help with testing and updating.  I ask you to do the same thing you asked me to do, that is if you can't have something your way fine, but don't get upset and mean about it.  Seems you could still be nice.  Dave worked to hard on his own time for us to have this framework.

**Comment by LuckyCyborg** [comment]
@jimgwhit Worry not, I'm really NOT upset about that "the super big giant monstor" I came up with was not adopted as Nova (v3). In fact, both my job tasks assigned about **SMVC** was reached.

As in pushing forward SMVC (Nova use **our** Routing to do its stunts and I still maintain that) and to develop (and contribute back) a SMVC based/inspired framework useful for our personal/in-house projects (the "monster" still kicking into **3.0-nova** branch). Why I should be upset?

To be honest, still I'm slightly disappointed about the daemon-ization which the 3.0-beta/nova received, even if we tried our best to preserve the code compatibility with 2.2 version. 

See that your own "the super big giant monstor", for example...

But, after all, was a very instructive journey in the Open-Source World for me, a novice into.

Finally, like @Kingorgg said, it is your business on how you use the framework, but to not yell tomorrow that you kicked mysterious security or code issues. Be warned for.

And, again, your suggestion/solution is **not a valid one**, from reasons which I said in a previous post. Trust me, I known well what should be changed to get ridding of Routing Assets Serving.

As bottom line, if you ask me to help you, how I can do that, if you don't consider my guidance?
